### ISSUE TITLE
Yii2 Error Handler with PHP 8.1

### ISSUE THREAD

**Issue by developedsoftware** [yiisoft/yii2#19340]
I have just updated to PHP 8.1.4 - and have a fatal error being generated due to method incompatibilty. In PHP 7.4 I would get an error displayed similar to the attached. 

![featured2](https://user-images.githubusercontent.com/19648131/162270027-017ee7c7-0d52-443e-8564-16f63ac971d2.jpg)

With 8.1 I just get a white screen. I have turned on error handling, YII_DEBUG etc as per the guide but the only way I seem to be able to get any sort of debug info out is by registering my own shutdown handler and dumping out error_get_last

Is this a known issue? Any pointers before I edit this and submit it as a proper issue?

**Comment by samdark** [comment]
Could it be reproduced with `new UnknownClass()` on a [basic application template](https://github.com/yiisoft/yii2-app-basic)?

**Comment by developedsoftware** [comment]
<img width="1492" alt="Screenshot 2022-04-07 at 22 59 46" src="https://user-images.githubusercontent.com/19648131/162326856-405fe32b-d924-49ee-b8dc-331e852a3ae7.png">

That renders fine! 

I will have to do some digging to try and pin it down. The error is inside a trait so maybe thats the key!

**Comment by mikehaertl** [comment]
Your screenshot shows that you have Yii 2.0.10. You should upgrade to 2.0.45. Versions before are not ready for PHP 8.1

**Comment by developedsoftware** [comment]
That was just a screen grab I found on the web (as I couldn’t get one to print out as it was white screening on me)

**Comment by mikehaertl** [comment]
I did a quick test with PHP 8.1.4 and Yii 2.0.45 using `$t = UnknownObject()` and can see the usual error page. So I'd say it must be somewhere in your config.

**Comment by developedsoftware** [comment]
It turned out to be a incompatible method declaration inside a trait. So I am not sure if yii is able to handle that as I guess it’s run at compile time rather than execution time.  My guess is that it cannot handle it and therefore the white screen is produced?

Once I fixed that the normal error style showed OK. 

But in order to find the method that was white screening I had to dump out error_get_last inside a shutdown handler.

**Comment by samdark** [comment]
So... how to reproduce it with a basic application template?

**Comment by developedsoftware** [comment]
Would you like me to create a repository demonstrating the issue or just explain how to re-create here ?

**Comment by samdark** [comment]
Just explaining is OK since it should be simple.

**Comment by WinterSilence** [comment]
you can't catch fatal error

**Comment by developedsoftware** [comment]
You can if you register a shutdown handler and check if error_get_last is null.

I will try to explain how I was able to reproduce the issue later today

**Comment by bizley** [comment]
@developedsoftware any news on this or can I close it?

**Comment by developedsoftware** [comment]
Cannot seem to replicate now, it works as expected. So yes will close.

**Comment by developedsoftware** [comment]
Ah! This was only added 14 days ago right? That makes sense that it works fine now

**Comment by WinterSilence** [comment]
@developedsoftware no, long time ago

**Comment by WinterSilence** [comment]
@developedsoftware it's can be opcache backside effect

**Comment by developedsoftware** [comment]
Yes it must have been as the same thing is not erroring any more. Sorry for wasting everyones time!
